src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,29 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  setSelectedTodo: (value: null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, setSelectedTodo }) => {
+  const [loading, setLoading] = useState(true);
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    getUser(todo.userId)
+      .then(setUser)
+      .finally(() => setLoading(false));
+  }, [todo.userId]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {loading ? (
```

Comment:
I don't quite like this ternary, so for the future you may consider following:

```suggestion
    if (loading) {
      return <Loader />
    }
    
    return (
      <div className="modal-card">
       ...
       ... render your modal in normal mode
```